src/picturepark-sdk-v2-angular/projects/picturepark-sdk-v2-angular/src/lib/locale.module.ts

```diff
@@ -20,7 +21,7 @@ export function loadLanguagesFactory(
 ): () => Promise<boolean | undefined> {
   const languageToSelect = language || getLocaleFactory(localStorageService);
   return () => {
-    return languageService.loadLanguages(allowedLanguages, languageToSelect, cdnUrl).toPromise();
+    return lastValueFrom(languageService.loadLanguages(allowedLanguages, languageToSelect, cdnUrl));
```

Comment:
add take(1) to ensure it's properly disposed

Reply:
No need to (and you can't pipe anything) as this doesn't return an observable but a promise.
According to this page (https://indepth.dev/posts/1287/rxjs-heads-up-topromise-is-being-deprecated) the now deprecated toPromise() function resolves once the observable completes. Therefore the lastValueFrom is the appropriate replacement as this one also resolves when the observable completes. firstValueFrom resolves on next() call